### PR TITLE
Migrate the FlutterSampleNotificationProvider to the new EditorNotificationProvider API (#7840)

### PR DESCRIPTION
**Pull request by jwren**
This is progress on https://github.com/flutter/flutter-intellij/issues/7830

### CODE REVIEW

File: flutter-idea/src/io/flutter/samples/FlutterSampleNotificationProvider.java [hunk 1]
```diff
@@ -6,87 +6,98 @@
...
 
     // Run the code to query the document in a read action.
     final List<FlutterSample> samples = ApplicationManager.getApplication().
       runReadAction((Computable<List<FlutterSample>>)() -> {
-        //noinspection CodeBlock2Expr
         return getSamplesFromDoc(flutterPackagePath, document, filePath);
       });
 
-    return samples.isEmpty() ? null : new FlutterSampleActionsPanel(samples);
+    if (samples != null && !samples.isEmpty()) {
```
Comment (pq) [inline flutter-idea/src/io/flutter/samples/FlutterSampleNotificationProvider.java:94]: Can `samples` be null here or can you skip this check? (It looks like `getSamplesFromDoc` at worst returns an empty list?)

File: flutter-idea/src/io/flutter/samples/FlutterSampleNotificationProvider.java [hunk 2]
```diff
@@ -6,87 +6,98 @@
...
+      return new FlutterSampleActionsPanel(samples);
+    }
+    return null;
   }
 
-  private List<FlutterSample> getSamplesFromDoc(String flutterPackagePath, Document document, String filePath) {
+  private List<FlutterSample> getSamplesFromDoc(@NotNull String flutterPackagePath, @NotNull Document document, @NotNull String filePath) {
```
Comment (pq) [inline flutter-idea/src/io/flutter/samples/FlutterSampleNotificationProvider.java:100]: Can you mark this `@NotNull`?

**Review by pq — APPROVED** [review]

File: flutter-idea/src/io/flutter/samples/FlutterSampleNotificationProvider.java (hunk 2 above)
Comment (jwren) [reply flutter-idea/src/io/flutter/samples/FlutterSampleNotificationProvider.java:100]: Done.

File: flutter-idea/src/io/flutter/samples/FlutterSampleNotificationProvider.java (hunk 1 above)
Comment (pq) [reply flutter-idea/src/io/flutter/samples/FlutterSampleNotificationProvider.java:94]: Ah! So maybe this is new Java. I would have expected `ApplicationManager.getApplication(). runReadAction` to throw an NPE if `getApplication` returns null, but cool!

**Pull request merged**